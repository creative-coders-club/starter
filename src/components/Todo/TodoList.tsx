import { Todo } from './Todo';
import { TodoType } from '../../types';

export const TodoList = ({ todos, removeTodo }: TodoListProps) => {
  return (
    <ul>
      {todos.map((todo) => (
        <Todo key={todo.id} todo={todo} removeTodo={removeTodo} />
      ))}
    </ul>
  );
};

interface TodoListProps {
  todos: TodoType[];
  removeTodo: (id: number) => void;
}
