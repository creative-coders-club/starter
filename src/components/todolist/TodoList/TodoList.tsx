import { TodoItem } from 'src/components/todolist/TodoItem';

import * as S from './TodoList.styles';

type TodoListProps = {
  todos: Todo[];
  onRemoveTodo: (todoId: Todo['id']) => void;
  onToggleTodoDone: (todoId: Todo['id']) => void;
};

// GYU-TODO: 테스트 코드 없어도 커버리지 100, 음.. 기능이 없고 UI 만 있어서 그런가?
export function TodoList({ todos, onRemoveTodo, onToggleTodoDone }: TodoListProps) {
  return (
    <S.TodoListWrapper>
      {todos.map((todo) => {
        return (
          <TodoItem
            key={todo.id} //
            todo={todo}
            onRemoveTodo={onRemoveTodo}
            onToggleTodoDone={onToggleTodoDone}
          />
        );
      })}
    </S.TodoListWrapper>
  );
}
