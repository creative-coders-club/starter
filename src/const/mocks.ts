export const mockData = [
  {
    id: 1,
    text: 'test1',
  },
];
