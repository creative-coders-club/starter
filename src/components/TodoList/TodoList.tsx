import { Todo } from '../Todo/Todo';
import { TodoType } from '../../types';
import S from '../TodoPage/TodoPage.module.scss';

export const TodoList = ({ todos, removeTodo }: TodoListProps) => {
  if (!todos.length) {
    return (
      <div className={S.todo_list_empty}>
        <p>There are no todos yet.</p>
      </div>
    );
  }

  return (
    <ul className={S.todo_list}>
      {todos.map((todo) => (
        <Todo key={todo.id} todo={todo} removeTodo={removeTodo} />
      ))}
    </ul>
  );
};

interface TodoListProps {
  todos: TodoType[];
  removeTodo: (id: number) => void;
}
