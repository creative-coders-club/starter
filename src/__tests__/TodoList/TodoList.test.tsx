import { render, screen } from '@testing-library/react';
import { TodoList } from '../../components/TodoList/TodoList';
import { mockData } from '../../const/mocks';

describe('TodoList 컴포넌트 테스트', () => {
  const { asFragment } = render(<TodoList todos={mockData} removeTodo={jest.fn()} />);
  it('TodoList가 전달받은 list를 잘 그려주는가', () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it('빈배열을 받았다면 There are no todos yet. 문구 표시되어야함', () => {
    render(<TodoList todos={[]} removeTodo={jest.fn()} />);

    // screen.getByText('There are no todos yet.')는 꼭 따로 변수에 담지 않아도 되는게 아닐까?
    // 오히려 더 선언적인거같기도?
    expect(screen.getByText('There are no todos yet.')).toBeInTheDocument();
  });
});
