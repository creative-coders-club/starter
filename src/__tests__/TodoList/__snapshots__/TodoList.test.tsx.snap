// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList 컴포넌트 테스트 TodoList가 전달받은 list를 잘 그려주는가 1`] = `
<DocumentFragment>
  <ul
    class="todo_list"
  >
    <li>
      test1
      <button
        class="todo_delete"
      >
        Delete
      </button>
    </li>
  </ul>
</DocumentFragment>
`;
